src/scheduler/workers/main.ts

```diff
@@ -144,49 +149,50 @@ export async function processList(list: List, listItemsForList: ListItemWithHist
   const audit = await getLatestReminderAuditEvent(annualReviewRef, "list");
   const listItemAudit = await getLatestReminderAuditEvent(annualReviewRef, "listItem");
   let isEmailSent = false;
-  const todayDateString = SCHEDULED_PROCESS_TODAY_DATE;
-  const today = new Date(todayDateString);
+
+  const today = getTodayDate();
   logger.info(`Checking annual review milestone dates against today date ${today.toISOString()} - ${JSON.stringify(annualReviewKeyDates)}`);
 
-  // email the posts and providers if today = one of the annual review milestone date
-  switch (today.toISOString()) {
-    case annualReviewKeyDates?.POST_ONE_MONTH:
-      isEmailSent = isEmailSentBefore(audit, "sendOneMonthPostEmail");
-      if (!isEmailSent) {
-        await processPostEmailsForList(list, "POST_ONE_MONTH", "sendOneMonthPostEmail");
-      }
-      break;
-    case annualReviewKeyDates?.POST_ONE_WEEK:
-      isEmailSent = isEmailSentBefore(audit, "sendOneWeekPostEmail");
-      if (!isEmailSent) {
-        await processPostEmailsForList(list, "POST_ONE_WEEK", "sendOneWeekPostEmail");
-      }
-      break;
-    case annualReviewKeyDates?.POST_ONE_DAY:
-      isEmailSent = isEmailSentBefore(audit as Audit, "sendOneDayPostEmail");
-      if (!isEmailSent) {
-        await processPostEmailsForList(list, "POST_ONE_DAY", "sendOneDayPostEmail");
-      }
-      break;
-    case annualReviewKeyDates?.START:
-      // email posts to notify of annual review start
-      isEmailSent = isEmailSentBefore(audit as Audit, "sendStartedPostEmail");
-      if (!isEmailSent) {
-        await processPostEmailsForList(list, "START", "sendStartedPostEmail");
-      }
+  if (isWithinInterval(today,{ start: annualReviewKeyDates?.POST_ONE_MONTH ?? "", end: annualReviewKeyDates?.POST_ONE_WEEK ?? "" })) {
```

Comment:
the ranges are probably from 00:00 to 00:00, right? is it worth end date being 23:59(:599)?

https://date-fns.org/v2.29.3/docs/endOfDay

I mentioned using isWithinInterval from date fns, but you may need to modify it.

 
```
import datefns from "date-fns"

// the type definition of datefns.isWithinInterval is isWithinInterval(date: Date | number, interval: Interval)

export function isWithinInterval(date: Date | number, interval: Interval) {
  if(!interval.start || !interval.end) {
    // log or throw, then return
  }

  // convert iso to date if you need to

  const intervalWithStartAndEndTime = {
     start: datefns.startOfDay(start),
     end: datefns.endOfDay(end),
  }
   
  return datefns.isWithinInterval(today, intervalWithStartAndEndTime);

}

```


I would probably even destructure all of these dates to help with readability. And then you can easily default without it adding too much noise to the interval checking code. 

```

const { 
  POST_ONE_MONTH = "",
  POST_ONE_WEEK = "",
  POST_ONE_DAY = "",
  POST_START = ""
} = annualReviewKeyDates ?? {};

if(isWithinInterval(today, { start: POST_ONE_MONTH, end: POST_ONE_WEEK }) { }

```

Reply:
makes sense but I'll leave this for another time.  I'm using endOfDay for the end date for now